plugins/main/public/components/common/search-bar/search-bar-service.ts

```diff
@@ -1,114 +1,148 @@
-    const searchSource = await data.search.searchSource.create();
-    const fromField = (pagination?.pageIndex || 0) * (pagination?.pageSize || 100);
-    const sortOrder: OpenSearchQuerySortValue[] = sorting?.columns.map((column) => {
-        const sortDirection = column.direction === 'asc' ? 'asc' : 'desc';
-        return { [column?.id || '']: sortDirection } as OpenSearchQuerySortValue;
+export const search = async (
+  params: SearchParams,
+): Promise<SearchResponse | void> => {
+  const {
+    indexPattern,
+    filters: defaultFilters = [],
+    query,
+    pagination,
+    sorting,
+    fields,
+    aggs,
+  } = params;
+  if (!indexPattern) {
+    return;
+  }
+  const data = getPlugins().data;
+  const searchSource = await data.search.searchSource.create();
+  const fromField =
+    (pagination?.pageIndex || 0) * (pagination?.pageSize || 100);
+  const sortOrder: OpenSearchQuerySortValue[] =
+    sorting?.columns.map(column => {
+      const sortDirection = column.direction === 'asc' ? 'asc' : 'desc';
+      return { [column?.id || '']: sortDirection } as OpenSearchQuerySortValue;
     }) || [];
-    let filters = defaultFilters;
+  let filters = defaultFilters;
 
-    // check if dateRange is defined
-    if (params.dateRange && params.dateRange?.from && params.dateRange?.to) {
-        const { from, to } = params.dateRange;
-        filters = [
-            ...filters,
-            {
-                // @ts-ignore
-                range: {
-                    [indexPattern.timeFieldName || 'timestamp']: {
-                        gte: from,
-                        lte: to,
-                        format: 'strict_date_optional_time'
-                    }
-                }
-            }
-        ]
+  // check if dateRange is defined
+  if (params.dateRange && params.dateRange?.from && params.dateRange?.to) {
+    // when the from and to are equal, the to is now
+    /* ToDo: Search the OSD date range parser, to prevent this change and fix
+        when the fllter is "Today" and "This week". In those ranges the filter to and from are the same
+        Today: from: now/d, to: now/d
+        This Week: from: now/w to:now/w
+    */
+    if (params.dateRange.from === params.dateRange.to) {
+      params.dateRange.to = 'now';
     }
+    const { from, to } = params.dateRange;
 
-    const searchParams = searchSource
-        .setParent(undefined)
-        .setField('filter', filters)
-        .setField('query', query)
-        .setField('sort', sortOrder)
-        .setField('size', pagination?.pageSize)
-        .setField('from', fromField)
-        .setField('index', indexPattern)
+    filters = [
+      ...filters,
+      {
+        // @ts-ignore
+        range: {
+          [indexPattern.timeFieldName || 'timestamp']: {
+            gte: dateMath.parse(from).toISOString(),
+            lte: dateMath.parse(to).toISOString(),
+            format: 'strict_date_optional_time',
+          },
+        },
```

Comment:
**thought**: the vulnerabilities states index pattern does not have timefield. Are we considering this case?

Reply:
I added a comment in the search-bar-service file explaining how to parse correctly the date range received by the search bar date picker.